File: app/src/main/java/eu/duboucher/mtbs0wn/MainActivity.java
package eu.duboucher.mtbs0wn;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.MotionEvent;
import android.widget.ImageView;
import android.widget.TextView;

import java.util.logging.Logger;

public class MainActivity extends AppCompatActivity {
	private final static Logger LOGGER = Logger.getLogger(Logger.GLOBAL_LOGGER_NAME);

	private void updateReservationCode() {
		ReservationCode _newReservationCode = ReservationCode.createReservationCode();

		_newReservationCode.updateTextView((TextView) findViewById(R.id.reservation_code));
		_newReservationCode.updateImageView((ImageView) findViewById(R.id.reservation_qrcode));
	}

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.activity_main);
		Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
		setSupportActionBar(toolbar);

		updateReservationCode();
	}

	@Override
	public boolean onTouchEvent(MotionEvent event) {
		switch (event.getActionMasked()) {
			case MotionEvent.ACTION_UP:
				updateReservationCode();
				break;
		}

		return super.onTouchEvent(event);
	}
}

File: app/src/main/kotlin/eu/duboucher/mtbs0wn/MainActivity.kt
package eu.duboucher.mtbs0wn

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.Toolbar
import android.view.MotionEvent
import android.widget.ImageView
import android.widget.TextView

class MainActivity:AppCompatActivity() {
  private fun updateReservationCode() {
    val _newReservationCode = ReservationCode.createReservationCode()
    _newReservationCode.updateTextView(findViewById(R.id.reservation_code) as TextView)
    _newReservationCode.updateImageView(findViewById(R.id.reservation_qrcode) as ImageView)
  }

  override protected fun onCreate(savedInstanceState : Bundle?) {
    super.onCreate(savedInstanceState)

    setContentView(R.layout.activity_main)

    val toolbar = findViewById(R.id.toolbar) as Toolbar
    setSupportActionBar(toolbar)

    updateReservationCode()
  }

  override fun onTouchEvent(event : MotionEvent) : Boolean {
    when (event.getActionMasked()) {
      MotionEvent.ACTION_UP -> updateReservationCode()
    }

    return super.onTouchEvent(event)
  }
}